content/Encrypting files with Active Storage and S3 Bucket Server-Side Encryption - Agustin Martinez.md

```diff
@@ -0,0 +1,204 @@
+# Encrypting files with Active Storage and S3 Bucket Server-Side Encryption
+
+In this article, we are going to see how encrypt/decrypt the files that we storage at Amazon S3 Bucket in our Rails API using Active Storage to manage our files.
+
+## Why we need to encrypt the files at S3 Bucket
+
+This requirement born from a client who wants to ensure the security of their clients' files which they upload or consume on the web site.
+
+Analyzing the context of the requirement, the first option that the team suggests was E2EE (end-to-end encryption). But this approach has a critical point: the user needs to share an encryption key between them, so, according to the user target of the business and the complex user experience, this option was discarded.
+
+Finally, the best option was to use an AWS Key to encrypt the files when the users upload it and decrypt when the user wants to view or play the file on the web site. Following the **server side encryption** approach.
```

Comment:
I would try to list all the options that were considered to avoid mentioning the first one and then the final decision.

Reply:
Can you help me to remember the unmentioned options?

* encrypt files before sending them to s3 at our server
*